app/controllers/vaccines_controller.rb

```diff
@@ -79,57 +66,79 @@ def create
-    dose_number = params.permit(:dose_number)[:dose_number]
-    notes = params.permit(:notes)[:notes]
-    patient_id = params.permit(:patient_id)[:patient_id]&.to_i
-
-    # Check if Patient ID is valid
-    unless Patient.exists?(patient_id)
-      error_message = "Vaccination cannot be created for unknown monitoree with ID: #{patient_id}"
-      render(json: { error: error_message }, status: :bad_request) && return
-    end
-
-    # Check if user has access to patient
-    unless current_user.get_patient(patient_id)
-      error_message = "User does not have access to Patient with ID: #{patient_id}"
-      render(json: { error: error_message }, status: :forbidden) && return
-    end
-
-    # Get vaccine to update
-    vaccine = Vaccine.find_by(id: vaccine_id)
-
-    # Handle case where vaccine cannot be found
-    render(json: { error: "Vaccination with ID #{vaccine_id} cannot be found." }, status: :bad_request) && return unless vaccine
-
-    # Update the vaccine record
     update_params = {
-      group_name: group_name,
-      product_name: product_name,
-      administration_date: administration_date,
-      dose_number: dose_number,
-      notes: notes,
-      patient_id: patient_id
+      group_name: params.permit(:group_name)[:group_name],
+      product_name: params.permit(:product_name)[:product_name],
+      administration_date: params.permit(:administration_date)[:administration_date],
+      dose_number: params.permit(:dose_number)[:dose_number],
+      notes: params.permit(:notes)[:notes],
+      patient_id: @patient_id
     }
 
     # Handle vaccine update success or failure
-    if vaccine.update(update_params)
+    if @vaccine.update(update_params)
       # Create history item on successful update
       History.vaccination_edit(
-        patient: params.permit(:patient_id)[:patient_id],
+        patient: @patient_id,
         created_by: current_user.email,
-        comment: "User edited a vaccination (ID: #{vaccine_id})."
+        comment: "User edited a vaccination (ID: #{@vaccine.id})."
       )
     else
       # Handle case where vaccine update failed
       error_message = 'Vaccination was unable to be updated. '
-      error_message += "Errors: #{format_model_validation_errors(vaccine).join(', ')}" if vaccine&.errors
+      error_message += "Errors: #{format_model_validation_errors(@vaccine).join(', ')}" if @vaccine&.errors
+      render(json: { error: error_message }, status: :bad_request) && return
+    end
+  end
+
+  def destroy
+    if @vaccine.destroy
+      reason = params.permit(:delete_reason)[:delete_reason]
+      comment = "User deleted a vaccine (ID: #{@vaccine.id}"
+      comment += ", Vaccine Group: #{@vaccine.group_name}" unless @vaccine.group_name.blank?
+      comment += ", Product Name: #{@vaccine.product_name}" unless @vaccine.product_name.blank?
+      comment += ", Administration Date: #{@vaccine.administration_date}" unless @vaccine.administration_date.blank?
```

Comment:
date needs to be formatted for display

Reply:
good catch! https://github.com/SaraAlert/SaraAlert/pull/897/commits/d3fb565d771c96258400a2931f7c7c376f8b1470